src/HonzaBotner/Controllers/AuthController.cs

```diff
@@ -66,17 +65,28 @@ public async Task<ActionResult> Callback()
                 return Content(e.Message, "text/html");
             }
 
-            bool auth = await _authorizationService.AuthorizeAsync(accessToken, userName, userId, rolesPool);
-
-            return Ok(auth);
+            return Ok(await _authorizationService.AuthorizeAsync(accessToken, userName, userId, rolesPool) switch
+            {
+                IAuthorizationService.AuthorizeResult.OK => "Sucessfully authenticated.",
+                IAuthorizationService.AuthorizeResult.Failed => "Authentication failed.",
+                IAuthorizationService.AuthorizeResult.DifferentMember =>
+                    "Authentication failed because you are registered with another auth code or another user already uses your auth code.",
+                IAuthorizationService.AuthorizeResult.UserMapError =>
+                    "Authentication failed due to UserMap service failure.",
+                _ => throw new ArgumentOutOfRangeException()
+            });
```

Comment:
Much better than previous state. In future we would ideally want some html page.